src/System Application/App/File System/src/FileSystem/FileSystemImp.Codeunit.al

```diff
@@ -0,0 +1,251 @@
+        CheckInitialization();
+        exit(FileSystemConnector.DirectoryExists(CurrFileAccount."Account Id", Path));
+    end;
+
+    procedure DeleteDirectory(Path: Text)
+    begin
+        CheckPath(Path);
+        CheckInitialization();
+        FileSystemConnector.DeleteDirectory(CurrFileAccount."Account Id", Path);
+    end;
+
+    procedure PathSeparator(): Text
+    begin
+        exit('/');
+    end;
+
+    procedure CombinePath(Path: Text; ChildPath: Text): Text
+    begin
+        if Path = '' then
+            exit(ChildPath);
+
+        if not Path.EndsWith(PathSeparator()) then
+            Path += PathSeparator();
+
+        exit(Path + ChildPath);
+    end;
+
+    procedure GetParentPath(Path: Text) ParentPath: Text
+    begin
+        Path := Path.TrimEnd(PathSeparator());
+        if Path.TrimEnd(PathSeparator()).Contains(PathSeparator()) then
+            ParentPath := Path.Substring(1, Path.LastIndexOf(PathSeparator()));
+    end;
+
+    procedure SelectFolderUI(Path: Text; DialogTitle: Text): Text
+    var
+        FileAccountContent: Record "File Account Content";
+        FileAccountBrowser: Page "File Account Browser";
+    begin
+        CheckPath(Path);
+        CheckInitialization();
+
+        FileAccountBrowser.SetPageCaption(DialogTitle);
+        FileAccountBrowser.SetFileAccount(CurrFileAccount);
+        FileAccountBrowser.EnableDirectoryLookupMode(Path);
+        if FileAccountBrowser.RunModal() <> Action::LookupOK then
+            exit('');
+
+        FileAccountBrowser.GetRecord(FileAccountContent);
+        if FileAccountContent.Type <> FileAccountContent.Type::Directory then
+            exit('');
+
+        exit(CombinePath(FileAccountContent."Parent Directory", FileAccountContent.Name));
+    end;
+
+    procedure SelectFileUI(Path: Text; FileFilter: Text; DialogTitle: Text): Text
+    var
+        FileAccountContent: Record "File Account Content";
+        FileAccountBrowser: Page "File Account Browser";
+    begin
+        CheckPath(Path);
+        CheckInitialization();
+
+        FileAccountBrowser.SetPageCaption(DialogTitle);
+        FileAccountBrowser.SetFileAccount(CurrFileAccount);
+        FileAccountBrowser.EnableFileLookupMode(Path, FileFilter);
+        if FileAccountBrowser.RunModal() <> Action::LookupOK then
+            exit('');
+
+        FileAccountBrowser.GetRecord(FileAccountContent);
+        if FileAccountContent.Type <> FileAccountContent.Type::File then
+            exit('');
+
+        exit(CombinePath(FileAccountContent."Parent Directory", FileAccountContent.Name));
+    end;
+
+    procedure SaveFileUI(Path: Text; FileExtension: Text; DialogTitle: Text): Text
+    var
+        FileAccountContent: Record "File Account Content";
+        FileAccountBrowser: Page "File Account Browser";
+        FileName, FileNameWithExtenion : Text;
+        PleaseProvideFileExtensionErr: Label 'Please provide a valid file extension.';
+        FileNameTok: Label '%1.%2', Locked = true;
```

Comment:
Comments for %1 filename , %2 file prefix

Reply:
Rule [AA0470](https://learn.microsoft.com/de-de/dynamics365/business-central/dev-itpro/developer/analyzers/codecop-aa0470?wt.mc_id=d365bc_inproduct_alextension#reason-for-the-rule) doesn't apply here. It is locked.